### PR TITLE
CIP8 message signing

### PR DESCRIPTION
**Pull request by Sarmaad**
Building upon Keith's PR #91 

Tested compatibility between Lucid and Eternl.

DataSigningUtility to generate Key and Signature.

### CODE REVIEW

File: CardanoSharp.Wallet/Utilities/SignDataUtility.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        public static DataSignature SignData(ICoseSigner coseSigner, PrivateKey rootKey, string payload,
+            int addressIndex, NetworkType networkType = NetworkType.Mainnet)
+        {
+            // stake 
+            var (_, sPub) = rootKey.GetKeyPairFromPath($"m/1852'/1815'/0'/2/0", false);
+
+            // payment address at index
+            var (aPri, aPub) = rootKey.GetKeyPairFromPath($"m/1852'/1815'/0'/0/{addressIndex}", false);
+
```
Comment (nothingalike) [inline CardanoSharp.Wallet/Utilities/SignDataUtility.cs:21]: Any reason we can't pass the path in and use the path from `rootKey` has default. I could see scenarios where you have the account level key and would like to start there.

**Review by nothingalike — COMMENTED** [review]
Looks good overall!

File: CardanoSharp.Wallet/Utilities/SignDataUtility.cs (hunk 1 above)
Comment (Sarmaad) [reply CardanoSharp.Wallet/Utilities/SignDataUtility.cs:21]: I see what you mean, but I wouldn't know how to solve for the stake public key.
I would need to introduce a stake path and assume the private key can only be a root or account key.

The signature of the method would look like this:

```
public static DataSignature SignData(ICoseSigner coseSigner, PrivateKey privateKey,
    string payload, string path = "m/1852'/1815'/0'/0/0",
    NetworkType networkType = NetworkType.Mainnet)
{
    // stake 
    var (_, sPub) = privateKey.GetKeyPairFromPath($"??", false);

    // payment address at index
    var (aPri, aPub) = privateKey.GetKeyPairFromPath(path, false);
```

This is an alternative:

```
        public static DataSignature SignData(ICoseSigner coseSigner, string payload, 
            PublicKey stake, PrivateKey payment,
            NetworkType networkType = NetworkType.Mainnet)
        {
            var aPub = payment.GetPublicKey(false);
            var address = AddressUtility.GetBaseAddress(aPub, stake, networkType);

            var coseSign1 = coseSigner.BuildCoseSign1(payload.ToBytes(), payment, address: address.GetBytes());

            var key = new CIPs.CIP8.Models
                .CoseKey(
                    KeyType.Okp,
                    AlgorithmId.EdDsa,
                    CurveType.Ed25519,
                    verificationKey: aPub.Key);
            
            return new()
            {
                Signature = coseSign1.GetCbor().EncodeToBytes().ToStringHex(),
                Key = key.GetCbor().EncodeToBytes().ToStringHex()
            };
        }
```

Comment (nothingalike) [reply CardanoSharp.Wallet/Utilities/SignDataUtility.cs:21]: > I see what you mean, but I wouldn't know how to solve for the stake public key.

I would have that as a parameter as well. so `paymentPath` and `stakePath` both with default values. 

as for the alternative, im always infavor of overloads that support different scenarios.

Comment (Sarmaad) [reply CardanoSharp.Wallet/Utilities/SignDataUtility.cs]: I have updated the code as discussed and included some documentation on the methods.

Also, while testing the hashed payload scenario, I found an issue in the CIP30 CoseSign1Extensions when the payload is hashed. The verification always fails.

Let me know if there is anything else.

**Review by nothingalike — APPROVED** [review]
LGTM 🌮

**Pull request merged**